WordPress/wordpress-playground
# [Blueprints] setSiteLanguage step – download the latest RC translations for Nightly and Beta builds of WordPress

This PR enables using `setSiteLanguage` with beta and nightly WordPress versions:

```json
{
	"preferredVersions": {
		"php": "8.0",
		"wp": "beta"
	},
	"steps": [
		{
			"step": "setSiteLanguage",
			"language": "en_GB"
		}
	]
}
```

Before this PR, the above Blueprint would fetch translations from `https://downloads.wordpress.org/translation/core/6.7-RC2/es_ES.zip` and fail after a 404 API response from the WordPress.org translations API. The `6.7-RC2` version string in the URL came directly from WordPress's version.php.

The WordPress translations API only offers translations for full, official WordPress releases (e.g. `6.6`, `6.6.1`)
plus RC translations for each major WordPress version (e.g. `6.6-RC`, `6.7-RC`). Note that beta and RC versions use the same translations bundle. Also, there is no translation bundle for the development versions so the best we can do is download the latest RC translations.

This PR transforms the raw version string, such as `6.7-RC2` or `6.8-alpha-59341`, into one recognized by the API, such as `6.7-RC` or `6.8-RC`. As a result, the above Blueprint now fetches the translations from `https://downloads.wordpress.org/translation/core/6.7-RC/es_ES.zip`.

## Testing Instructions (or ideally a Blueprint)

- CI

### Manual testing instructions

- Open Nightly in Spanish http://127.0.0.1:5400/website-server/?wp=nightly&language=es_ES&url=%2Fwp-admin%2F
- Confirm WP-admin is in Spanish
- Open Beta in Spanish  http://127.0.0.1:5400/website-server/?wp=beta&language=es_ES&url=%2Fwp-admin%2F
- Confirm WP-admin is in Spanish
- Open 6.5 in Spanish http://127.0.0.1:5400/website-server/?wp=6.5&language=es_ES&url=%2Fwp-admin%2F
- Confirm WP-admin is in Spanish
